framework/data/ArrayDataProvider.php

```diff
@@ -63,6 +63,11 @@ class ArrayDataProvider extends BaseDataProvider
      * The array elements must use zero-based integer keys.
      */
     public $allModels;
+    /**
+     * @var bool If the $allModels need is already part of the data set the value to false
```

Comment:
It took much time for me to understand, what does it do. I suggest to change description to

```
@var boolean whether we need to slice [[allModels]] for pagination. Set to `false`, if [[allModels]] are already paginated. Works only when [[pagination]] is set. Defaults to `true`.
```